### PR TITLE
Implement concat

### PR DESCRIPTION
**Pull request by BruceDai**
@huningxin PTAL, thanks.

### CODE REVIEW

File: src/concat.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          }
+        } else {
+          outputShape[axis] += shape[axis];
+        }
+      }
+    }
+  }
+  const output = new Tensor(outputShape);
+  let outputIndex = 0;
+  const times = sizeOfShape(inputShape.slice(0, axis));
```
Comment (huningxin) [inline src/concat.js:41]: what does `times` mean?

File: src/concat.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  for (let t = 0; t < times; ++t) {
+    for (let k = 0; k < inputs.length; ++k) {
+      const width = sizeOfShape(inputs[k].shape.slice(axis));
+      for (let w = 0; w < width; ++w) {
+        output.setValueByIndex(outputIndex++, inputs[k].getValueByIndex(t * width + w));
+      }
+    }
+  }
```
Comment (huningxin) [inline src/concat.js:49]: This algorithm seems not so straight forward. It actually partitions the tensors to 2-D and concatenates.

How about iterating through the output elements and pick the input from inputs according to the output location (dim at output.shape[axis])?

e.g.
```js
  for (let i = 0; i < sizeOfShape(output.shape); ++i) {
    const location = output.locationFromIndex(i);
    let dim = location[axis];
    let k = 0;
    // Find out input k and its dim of axis according to output dim of axis
    for (; k < inputs.length; ++k) {
      if (dim < inputs[k].shape[axis]) {
        break;
      }
      dim -= inputs[k].shape[axis];
    }
    location[axis] = dim;
    const inputValue = inputs[k].getValueByLocation(location);
    output.setValueByIndex(i, inputValue);
  }
```

Comment (BruceDai) [reply src/concat.js:49]: Yes, my algorithm isn't  straight forward, it's to flat all inputs to 1-D tensors, then use the slide-window whose length is `sizeOfShape(inputs[k].shape.slice(axis))` to get input value one by one to fill the output value buffer.

Thanks @huningxin. I will replace the algorithm with yours.

**Comment by BruceDai** [comment]
@huningxin Please take a look again, thanks.

File: src/concat.js (hunk 2 above)
Comment (huningxin) [reply src/concat.js:54]: thanks!

**Review by huningxin — APPROVED** [review]

**Pull request merged**